### ISSUE TITLE
Wrong implementation of interface for Magento\Shipping\Model\Carrier\AbstractCarrier

### ISSUE THREAD

**Issue by joshuaadickerson** [magento/magento2#19592]
### Preconditions (*)

1. 2.2.6
2. 2.2.7

### Steps to reproduce (*)

1. Start with a version <= 2.2.5
2. Extend AbstractCarrier
3. Override `proccessAdditionalValidation` since that was the only method available at the time
4. Upgrade to 2.2.6 or 2.2.7

### Expected result (*)

1. The caller should call `proccessAdditionalValidation` or AbstractCarrier should internally call `proccessAdditionalValidation` from `processAdditionalValidation`
2.

### Actual result (*)

1. Your class breaks because `proccessAdditionalValidation` never gets called. The caller calls the new, correctly named, method, `processAdditionalValidation` and never calls the old one.

https://magentocommeng.slack.com/archives/C4YS78WE6/p1543952275173500
https://magentocommeng.slack.com/archives/C4YS78WE6/p1543952325174500
https://magentocommeng.slack.com/archives/C4YS78WE6/p1543952435175500

This was broken in https://github.com/magento/magento2/pull/16414

**Comment by magento-engcom-team** [comment]
Hi @joshuaadickerson. Thank you for your report. 
To help us process this issue please make sure that you provided the following information:
- [ ] Summary of the issue
- [ ] Information on your environment
- [ ] Steps to reproduce
- [ ] Expected and actual results

Please make sure that the issue is reproducible on the vanilla Magento instance following **Steps to reproduce**. To deploy vanilla Magento instance on our environment, please, add a comment to the issue: 

`@magento-engcom-team give me $VERSION instance` 

where `$VERSION` is [version tags](https://github.com/magento/magento2/tags) (starting from 2.2.0+) or develop branches (for example: 2.3-develop). 
For more details, please, review the [Magento Contributor Assistant](https://devdocs.magento.com/guides/v2.2/contributor-guide/contributing.html#contributor-assist) documentation.

@joshuaadickerson do you confirm that you was able to reproduce the issue on vanilla Magento instance following steps to reproduce?
- [ ] yes
- [ ] no

**Comment by magento-engcom-team** [comment]
Hi @engcom-backlog-nazar. Thank you for working on this issue. 
In order to make sure that issue has enough information and ready for development, please read and check the following instruction: :point_down:
- [x] 1. Verify that issue has all the required information. (**Preconditions**, **Steps to reproduce**, **Expected result**, **Actual result**).<details><summary>Details</summary>If the issue has a valid description, the label `Issue: Format is valid` will be added to the issue automatically. Please, edit issue description if needed, until label `Issue: Format is valid` appears.</details>

- [x] 2. Verify that issue has a meaningful description and provides enough information to reproduce the issue. If the report is valid, add `Issue: Clear Description` label to the issue by yourself.

- [x] 3. Add `Component: XXXXX` label(s) to the ticket, indicating the components it may be related to.
- [x] 4. Verify that the issue is reproducible on `2.3-develop` branch<details><summary>Details</summary>- Add the comment `@magento-engcom-team give me 2.3-develop instance` to deploy test instance on Magento infrastructure. <br/> - If the issue is reproducible on `2.3-develop` branch, please, add the label `Reproduced on 2.3.x`.<br/>- If the issue is not reproducible, add your comment that issue is not reproducible and close the issue and **_stop verification process here_**!
</details>

- [x] 5. Verify that the issue is reproducible on `2.2-develop` branch. <details><summary>Details</summary>- Add the comment `@magento-engcom-team give me 2.2-develop instance` to deploy test instance on Magento infrastructure. <br/>- If the issue is reproducible on `2.2-develop` branch, please add the label `Reproduced on 2.2.x`</details>

- [x] 6. Add label `Issue: Confirmed` once verification is complete. 

- [x] 7. Make sure that automatic system confirms that report has been added to the backlog.

**Comment by magento-engcom-team** [comment]
@engcom-backlog-nazar Thank you for verifying the issue. Based on the provided information internal tickets `MAGETWO-96977`, `MAGETWO-96978` were created

**Comment by orlangur** [comment]
Another related PR: https://github.com/magento/magento2/pull/19314/files

Looks like in this case keeping a method with a typo is a lesser evil :(

**Comment by magento-engcom-team** [comment]
@joshuaadickerson, thank you for your report.     
    
Unfortunately, we are archiving this ticket now as it did not get much attention from both Magento Community and Core developers for an extended period. This is done in an effort to create a quality, community-driven backlog which will allow us to allocate the required attention more easily.

Please feel free to comment, reopen or create new ticket according to the [Issue reporting guidelines](https://github.com/magento/magento2/wiki/Issue-reporting-guidelines) 
if you are still facing this issue on the latest `2.3-develop` branch. Thank you for collaboration.
